### PR TITLE
Implements some `shows` endpoints

### PR DESCRIPTION
**Pull request by hayribakici**
This PR implements some endpoints regarding the shows api:

* save shows for the current user
* remove shows for the current user
* checking whether user saved shows.

This PR also depracates `isSavedAlbums`. A new method `containsSavedAlbums` replaces it, since it returns a `Map<String, bool>` with the key as the `album id` and value whether the id has been saved or not.

### CODE REVIEW

File: lib/src/endpoints/me.dart [hunk 1]
```diff
@@ -155,9 +155,41 @@ class Me extends EndpointPaging {
...
+  /// [market] - An ISO 3166-1 alpha-2 country code. If a country code is
+  /// specified, only content that is available in that market will be returned.
+  Future<void> removeShows(List<String> ids, [String market = '']) async {
+    assert(ids.isNotEmpty, 'No show ids were provided for removing');
+    var query = _buildQuery({'ids': ids.join(','), 'market': market});
+    await _api._delete('$_path/shows?' + query);
+  }
+
+  /// Check if passed albums (ids) are saved by current user. The output
+  /// [ids] the ist of id's to check
```
Comment (rinukkusu) [inline lib/src/endpoints/me.dart:178]: The comments on this method seem to be a bit jumbled :)

**Review by rinukkusu — CHANGES_REQUESTED** [review]
Looks good otherwise!

File: lib/src/endpoints/me.dart (hunk 1 above)
Comment (hayribakici) [reply lib/src/endpoints/me.dart:178]: indeed 😅

Comment (rinukkusu) [reply lib/src/endpoints/me.dart:178]: Make it a "list" in line 178 and I'm happy 😆

Comment (hayribakici) [reply lib/src/endpoints/me.dart:178]: well this is what happens when you code late at night 😆

**Comment by hayribakici** [comment]
Regarding the `Future<Map<String, bool>> isSavedShows(List<String> ids)`. It returns a map with the given `id's` and the response whether the show has been saved or not. I was wondering if e.g. the `isSavedAlbums` method should also return a map with the given id's and the `bool` flag.

File: lib/src/endpoints/me.dart (hunk 1 above)
Comment (rinukkusu) [reply lib/src/endpoints/me.dart]: It happens 😄

**Comment by hayribakici** [comment]
In this PR?

**Comment by rinukkusu** [comment]
> In this PR?

Sure, why not!

> I would rather deprecate `isSavedAlbums` and create a new method with the `Map` to ensure backwards compatibilty.

Sounds like the safer approach, yes!

**Comment by rinukkusu** [comment]
LGTM! Thanks for the new features! :)
I'm fixing some new (very) pedantic warnings, then the workflow should be able to publish again!

**Review by rinukkusu — APPROVED** [review]

**Pull request merged**